tests/backends/test_utils.py

```diff
@@ -90,3 +91,23 @@ def test_unsupported_callproc_kparams_raises_error(self):
         with self.assertRaisesMessage(NotSupportedError, msg):
             with connection.cursor() as cursor:
                 cursor.callproc('test_procedure', [], {'P_I': 1})
+
+
+class CursorDebugWrapperTests(SimpleTestCase):
+    @mock.patch('django.db.backends.utils.logger')
+    def test_debug_sql_logging(self, logger):
+        cursor = mock.Mock()
+
+        # Create a mock database instance including the wrap_database_errors context manager.
+        db = mock.MagicMock(alias='mydb', execute_wrappers=[])
+        db.wrap_database_errors.__exit__.return_value = True
+
+        wrapper = CursorDebugWrapper(cursor, db)
+        wrapper.execute(sql='SELECT 1;')
+
+        call_args = logger.debug.call_args[0]
+        self.assertTrue('; alias=' in call_args[0])
+        self.assertEqual(db.alias, call_args[4])
+
+        extra = logger.debug.call_args[1]['extra']
+        self.assertEqual(db.alias, extra['alias'])
```

Comment:
IMO this tests should be moved to the `tests.backends.tests.BackendTestCase`, also we don't need to mock a database, checking for a `connection.alias` in a logger should be enough, e.g.
```diff
diff --git a/tests/backends/tests.py b/tests/backends/tests.py
index da20d94442..5f6e02d91d 100644
--- a/tests/backends/tests.py
+++ b/tests/backends/tests.py
@@ -3,6 +3,7 @@ import datetime
 import threading
 import unittest
 import warnings
+from unittest import mock
 
 from django.core.management.color import no_style
 from django.db import (
@@ -492,6 +493,23 @@ class BackendTestCase(TransactionTestCase):
             BaseDatabaseWrapper.queries_limit = old_queries_limit
             new_connection.close()
 
+    @mock.patch('django.db.backends.utils.logger')
+    @override_settings(DEBUG=True)
+    def test_queries_logger(self, mocked_logger):
+        sql = 'SELECT 1' + connection.features.bare_select_suffix
+        with connection.cursor() as cursor:
+            cursor.execute(sql)
+        params, kwargs = mocked_logger.debug.call_args
+        self.assertIn('; alias=%s', params[0])
+        self.assertEqual(params[2], sql)
+        self.assertEqual(params[3], None)
+        self.assertEqual(params[4], connection.alias)
+        self.assertEqual(
+            list(kwargs['extra']),
+            ['duration', 'sql', 'params', 'alias'],
+        )
+        self.assertEqual(tuple(kwargs['extra'].values()), params[1:])
+
     def test_timezone_none_use_tz_false(self):
         connection.ensure_connection()
         with self.settings(TIME_ZONE=None, USE_TZ=False):
```